src/rust/ide/view/graph-editor/src/component/node/vcs.rs

```diff
@@ -0,0 +1,193 @@
+//! Functionality related to visualising the version control system status of a node.
+
+use crate::prelude::*;
+use ensogl::display::shape::*;
+use crate::component::node as node;
+use ensogl::gui::component::ShapeView;
+use ensogl::display;
+use ensogl::application::Application;
+use enso_frp as frp;
+use ensogl::data::color;
+
+
+
+// ==============
+// === Status ===
+// ==============
+
+/// The version control system status of a node.
+#[derive(Debug,Copy,Clone)]
+pub enum Status {
+    /// The node has not been modified.
+    Unchanged,
```

Comment:
I don't like this design as it repeats the ideas of Option instead of reusing it. Instead I' create:

```rust
pub enum Change { Added, Edited };
pub type Status = Option<Change>;
```